### ISSUE TITLE
Edge Cases in ITRF to Geodetic

### ISSUE THREAD

**Issue by thkruz** [david-rc-dayton/pious_squid#2]
I was doing some edge case testing including ITRF positions at the north and south pole. When creating a position at the north pole and attempting to convert it to a geodetic position I am getting an incorrect altitude (negative radius of the earth).

https://github.com/david-rc-dayton/pious_squid/blob/d3579648da9be426c76c910e4c12c460171e86bd/lib/src/coordinate/itrf.dart#L54C3-L58C6

I had to write an explicit check for this condition:

```ts
if (x === 0 && y === 0) {
  lat = phi;
  alt = z > 0 ? ((z - Earth.radiusPolar) as Kilometers) : ((z + Earth.radiusPolar) as Kilometers);
} else {
// No changes to this branch
  for (let i = 0; i < 20; i++) {
    const slat = Math.sin(lat);
     c = 1 / Math.sqrt(1 - esq * slat * slat);
    lat = Math.atan((z + sma * c * esq * slat) / r);
  }
  alt = (r / Math.cos(lat) - sma * c) as Kilometers;
}
```

**Comment by david-rc-dayton** [comment]
@thkruz Can you post the typescript code that produces the incorrect value?

**Comment by thkruz** [comment]
No problem.

```js
const testDate= new Date(1705109326817);
const itrf = new ITRF(
  EpochUTC.fromDateTime(testDate),
  new Vector3D(0 as Kilometers, 0 as Kilometers, Earth.radiusPolar),
  new Vector3D(0 as KilometersPerSecond, 0 as KilometersPerSecond, 0 as KilometersPerSecond),
).toGeodetic().alt
```
It should be near 0. I was comfortable with some margin of error given radiusPolar vs radiusEquator, but it was not even close. When I stepped through the code in the debugger I realized that `final alt` on line 59 was clearly incorrect.

**Comment by david-rc-dayton** [comment]
Thanks, I was able to reproduce the issue!

I ended up swapping the altitude line with:
```dart
final alt = (pi * 0.5 - lat.abs() > (pi / 180))
        ? r / cos(lat) - c
        : z / sin(lat) - (c * (1 - esq));
```
...it seems to have fixed the issue. I'll need to do some more testing to make sure it doesn't break stuff.

**Comment by david-rc-dayton** [comment]
~Disregard the change. It flips the south pole, which is suboptimal.~

**Comment by david-rc-dayton** [comment]
Disregard the previous disregard. It works fine. I'm just bad at quickly throwing together tests.

**Comment by david-rc-dayton** [comment]
Fixed in `0.3.0` release.
